server/players.py

```diff
@@ -12,6 +12,7 @@ class PlayerState(IntEnum):
     HOSTING = 3,
     JOINING = 4,
     SEARCHING_LADDER = 5,
+    IN_PARTY = 6,
```

Comment:
I question the usefulness of this new disjoint state -- can a player not be both `PLAYING` and `IN_PARTY`?

I realize the same is true for `SEARCHING_LADDER`, but this smells like a clue of state-bloat. The primary reason this is here is to quickly determine valid protocol elements; e.g. no GameConnection messages are allowed if the state isn't either `PLAYING` or `HOSTING`. It seems like a bit-field/set of states is more appropriate though?

Reply:
I agree, there needs to be some work to be done on this.it's for blocking joining a game,... while in queue.

We should probably use something like INQUEUE. Although there are many open questions like what do we do with the group after they are in game/finished, do we kick the player from the grouo on joining a game or just Block queueing. And most importantly, on matching, do we ask players if the are ready?